core/corehttp/gateway_handler.go

```diff
@@ -338,10 +338,8 @@ func (i *gatewayHandler) putHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	tctx, cancel := context.WithTimeout(ctx, time.Minute)
-	defer cancel()
```

Comment:
im not sure this is a bad context to have-- what is the timeout on `ctx` as is? is there one?